src/Test.UnitTests.Sarif/FileEncodingTests.cs

```diff
@@ -89,12 +89,19 @@ public void FileEncoding_UnicodeDataIsTextual()
             using var assertionScope = new AssertionScope();
 
             var sb = new StringBuilder();
-            string unicodeText = "американец";
+            string[] unicodeTexts = new[]
+            {
+                "американец",
+                "Generates a warning and an error for each of :  foo  foo \r\n" // Challenging for the classifer; found by accident.
```

Comment:
\nNULL seems to break decoding as text for UTF16.  I did not fully isolate which part of the string failed us in UTF32 decoding, but I do know the classifier determines this thing to be binary for UTF32 due to the check on line 69 (new code), which is a block that I doubt we're hitting in any other way.

Reply:
Not sure what you mean. First is this data '\n\0' that you are referring to?

Second, any input fewer than 4 bytes will result in 0 characters read from the content, if it comes to it, correct?

I think I may be confused, reach out offline if it's faster.

Reply:
ok! Now I understanding, you are referring to the very final chars of the test string. :) Oops.